Loop/Managers/StatusExtensionDataManager.swift

```diff
@@ -0,0 +1,95 @@
+        guard let glucoseStore = self.dataManager.glucoseStore else {
+            completionHandler(nil)
+            return
+        }
+
+        dataManager.loopManager.getLoopStatus {
+            (predictedGlucose, _, recommendedTempBasal, lastTempBasal, lastLoopCompleted, _, _, error) in
+            
+            let context = StatusExtensionContext()
+        
+            #if IOS_SIMULATOR
+                // If we're in the simulator, there's a higher likelihood that we don't have
+                // a fully configured app. Inject some baseline debug data to let us test the
+                // experience. This data will be overwritten by actual data below, if available.
+                context.batteryPercentage = 0.25
+                context.reservoir = ReservoirContext(startDate: Date(), unitVolume: 42.5, capacity: 200)
+                context.netBasal = NetBasalContext(rate: 2.1, percentage: 0.6, startDate: Date() - TimeInterval(250))
+                context.eventualGlucose = String(
+                    format: NSLocalizedString("Eventually %@", comment: ""),
+                    String(describing: HKQuantity(unit: HKUnit.milligramsPerDeciliterUnit(), doubleValue: 119.0)))
+            #endif
+
+            let dataManager = self.dataManager
+            
+            context.loop = LoopContext(
+                dosingEnabled: dataManager.loopManager.dosingEnabled,
+                lastCompleted: lastLoopCompleted)
+
+            if let glucose = glucoseStore.latestGlucose {
+                context.latestGlucose = GlucoseContext(
+                    latest: glucose,
+                    sensor: dataManager.sensorInfo)
+            }
+            
+            let (netBasalRate, netBasalPercentage, basalStartDate) = dataManager.loopManager.calculateNetBasalRate()
+            if let rate = netBasalRate, let percentage = netBasalPercentage, let startDate = basalStartDate {
+                context.netBasal = NetBasalContext(rate: rate, percentage: percentage, startDate: startDate)
+            }
+            
+            if let reservoir = dataManager.doseStore.lastReservoirValue,
+               let capacity = dataManager.pumpState?.pumpModel?.reservoirCapacity {
+                context.reservoir = ReservoirContext(
+                    startDate: reservoir.startDate,
+                    unitVolume: reservoir.unitVolume,
+                    capacity: capacity)
+            }
+            
+            if let batteryPercentage = dataManager.latestPumpStatusFromMySentry?.batteryRemainingPercent {
+                context.batteryPercentage = Double(batteryPercentage) / 100.0
+            }
+        
+            if let lastPoint = predictedGlucose?.last {
+                context.eventualGlucose = String(
+                    format: NSLocalizedString("Eventually %@", comment: "The subtitle format describing eventual glucose. (1: localized glucose value description)"),
```

Comment:
sorry, I didn't follow your last comment - not sure if it was a gihub glitch, or did you cross comment streams? I think your comment might have been in regard to the linker warning we're seeing, but this comment stream is about the eventual glucose string.

re: eventual glucose string, one approach is to pull apart the HKQuantity into a double value and a unit string. Problem with that is that I can't seem to find any way to extract the unit type from the HKQuantity. I'll go ahead and do that and hardcode it to mg/dL for now and we'll do the formatting on the far side in the ViewController.